fondant/pipeline.py

```diff
@@ -0,0 +1,447 @@
+                                f"{subset_field}"
+                            )
+            load_component = False
+
+        logger.info("All pipeline component specifications match.")
+
+    def compile_pipeline(self):
+        """
+        Function that creates and compiles a Kubeflow Pipeline.
+        Once you have generated the pipeline function, you can use it to create an instance of
+        the pipeline and compile it using the Kubeflow compiler.
+        """
+
+        def _get_component_function(
+            fondant_component_operation: FondantComponentOp,
+        ) -> t.Callable:
+            """
+            Load the Kubeflow component based on the specification from the fondant component
+             operation.
+
+            Args:
+                fondant_component_operation (FondantComponentOp): The fondant component
+                 operation.
+
+            Returns:
+                Callable: The Kubeflow component.
+            """
+            return kfp.components.load_component(
+                text=fondant_component_operation.kubeflow_specification.to_string()
+            )
+
+        def _set_task_configuration(task, fondant_component_operation):
+            # Unpack optional specifications
+            number_of_gpus = fondant_component_operation.number_of_gpus
+            node_pool_name = fondant_component_operation.node_pool_name
+            p_volumes = fondant_component_operation.p_volumes
+            ephemeral_storage_size = fondant_component_operation.ephemeral_storage_size
+
+            # Assign optional specification
+            if number_of_gpus is not None:
+                task.set_gpu_limit(number_of_gpus)
+            if node_pool_name is not None:
+                task.add_node_selector_constraint("node_pool", node_pool_name)
+            if p_volumes is not None:
+                task.add_pvolumes(p_volumes)
+            if ephemeral_storage_size is not None:
+                task.set_ephemeral_storage_request(ephemeral_storage_size)
+
+            return task
+
+        # Sort graph based on specified dependencies
+        self.sort_graph()
+
+        # parse metadata argument required for the first component
+        run_id = "{{workflow.name}}"
+
+        # Validate subset schema before defining the pipeline
+        self._validate_pipeline_definition(run_id)
+
+        @dsl.pipeline(name=self.name, description=self.description)
+        def pipeline():
+            # TODO: check if we want to have the manifest path empty for loading component or remove
+            #  it completely from the loading component
+            # TODO: check if we want to have the metadata arg empty for transform component or
+            #  remove it completely from the transform component
+            manifest_path = ""
+            metadata = ""
+            previous_component_task = None
+            for _, operation_specs in self._graph.items():
```

Comment:
You can use `.values()` so you don't have the unused parameter. Also don't think `operation_specs` is the best name here. Maybe just `operation`.